library/lib_num_to_formatted_str.ks

```diff
@@ -48,23 +55,32 @@ LOCAL FUNCTION time_string {
     SET fixedPlaces TO maxLength.
   }
 
-  FROM {LOCAL i IS maxLength - fixedPlaces.} UNTIL i >= maxLength STEP {SET i TO i + 1.} DO {
-    SET returnString TO padding(timeList[i],lib_formatting_lex["leading0List"][i],roundingList[i],FALSE,1) + stringList[i] + returnString.
+  FROM {LOCAL i IS maxLength - fixedPlaces.}
+  UNTIL i >= maxLength STEP {SET i TO i + 1.} DO {
+    LOCAL leading0List IS lib_formatting_lex["leading0List"][i].
+    LOCAL number IS ROUND(timeList[i], roundingList[i]).
+    LOCAL pad_str IS padding(number, leading0List, roundingList[i], FALSE, 1).
```

Comment:
```suggestion
    LOCAL pad_str IS padding(number, leading0List, roundingList[i], FALSE, 0).
```
Adding a new line for rounding would not be the way to change how the function is rounding.  Instead changing the the augments passed into the padding function would be the better method.

Reply:
Since rounding can overflow and needs to "carry the one" I put a fix for this in time_converter. See d60e01b for the specific change.

Reply:
Having looked at your solution and through about the rounding problem a bit I think the better solution would be to round the `timeSec` value in `time_formatting` before it gets passed into `time_string`.  This should solve the rounding issue and avoid adding more complexity to the loop in `time_converter`.

Reply:
I'm looking at reverting d60e01b and doing the rounding in `time_formatting`. Would that nullify the need for `roundingList`?

Reply:
I used the `roundingList` because doing improved the performance of the concatenation loop in `time_string` than any other method I could think of at the time.  The only thing I can think of that might match it's performance in the concatenation loop would be a ternary operation in the call of the `padding` but doing that starts putting rather a large amount of logic in the `padding` call.

Reply:
Never mind , I see `roundingList` is used  in `time_string` to get trailing zeros for the seconds.

See 7fb417a for my implementation of your suggestion.
I think `time_string` is a better location for the rounding since it a lower level internal function.